### PR TITLE
feat(stonedb): show stonedb tag version. (#1251)

### LINKED ISSUE #1251: feature: show stonedb tag version
**Is your feature request related to a problem? Please describe.**

release version format：5.7.36-StoneDB-v1.0.3  
debug version format：5.7.36-StoneDB-v1.0.3.abcdefaaa  (abcdefaaa  means the first 9 digits of commit id string)
related：
1. mysql welcome page
2. select version()
3. mysqld --version
4. related logs

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: close #1251

## Tests Check List

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [X] New Feature
- [ ] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by lujiashun** [comment]
for debug version
```
root@ub01:/data/codebase/stonedb57/stonedb/build/build_stonedb-5.7-dev_x86_64_Ubuntu20.04_Debug_2023-01-30_12-48-37# /data/stonedb57/install/bin/mysql -uroot -p -S /data/stonedb57/install/tmp/mysql.sock
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 2
Server version: 5.7.36-StoneDB-v1.0.2.b1350c790 build-

Copyright (c) 2021, 2022 StoneAtom Group Holding Limited
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select version
    -> ^C
mysql> select version();
+---------------------------------+
| version()                       |
+---------------------------------+
| 5.7.36-StoneDB-v1.0.2.b1350c790 |
+---------------------------------+
1 row in set (0.00 sec)

mysql>

```

**Comment by lujiashun** [comment]
for release version:
```
root@ub01:/data/codebase/stonedb57/stonedb/build# /data/stonedb57/install/bin/mysql -uroot -p -S /data/stonedb57/install/tmp/mysql.sock
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 3
Server version: 5.7.36-StoneDB-v1.0.2 build-

Copyright (c) 2021, 2022 StoneAtom Group Holding Limited
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select version();
+-----------------------+
| version()             |
+-----------------------+
| 5.7.36-StoneDB-v1.0.2 |
+-----------------------+
1 row in set (0.00 sec)
```

**Review by wisehead — APPROVED** [review]

File: mysql-test/suite/tianmu/r/issue1251.result [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select (select version() regexp '^5.7.36-StoneDB-v[0-9]*\\.[0-9]*\\.[0-9]*$') or (select version() regexp '^5.7.36-StoneDB-v[0-9]*\\.[0-9]*\\.[0-9]*\\.[0-9a-z]{9}$');
```
Comment (RingsC) [inline mysql-test/suite/tianmu/r/issue1251.result:1]: if we upgrade our core to `5.7.4x`, what will happen to this line code.

Comment (lujiashun) [reply mysql-test/suite/tianmu/r/issue1251.result:1]: the sql result wil be 0, need to modify the testcase if upgrade to 5.7.4x

**Review by RingsC — APPROVED** [review]

File: sql/mysqld.cc [hunk 2]
```diff
@@ -8103,6 +8103,7 @@ static void set_server_version(void)
 {
   char *end= strxmov(server_version, MYSQL_SERVER_VERSION,
                      MYSQL_SERVER_SUFFIX_STR, NullS);
+#if 0
```
Comment (hustjieke) [inline sql/mysqld.cc:8108]: Add some comments here will be better? 
Old code logic --> new code.
`mysqld` include storage engines like `tianmu`, `innodb`, `myisam`..., after we change the code, `select version()` acctually only shows `tianmu` version.

Comment (lujiashun) [reply sql/mysqld.cc:8108]: OK, thanks, I will add some comment here later.  select version()  shows stonedb version,  include the sql layer and storage layer include tianmu, innodb

**Pull request merged**